pkg/build/util/util.go

```diff
@@ -162,3 +162,43 @@ func BuildDeepCopy(build *buildapi.Build) (*buildapi.Build, error) {
 	}
 	return copied, nil
 }
+
+// MergeTrustedEnvWithoutDuplicates merges two environment lists without having
+// duplicate items in the output list.  Only trusted environment variables
+// will be merged.
+func MergeTrustedEnvWithoutDuplicates(source []kapi.EnvVar, output *[]kapi.EnvVar, sourcePrecedence bool) {
+	// filter out all environment variables except trusted/well known
+	// values, because we do not want random environment variables being
+	// fed into the privileged STI container via the BuildConfig definition.
+	type sourceMapItem struct {
+		index int
+		value string
+	}
+
+	index := 0
+	filteredSourceMap := make(map[string]sourceMapItem)
+	filteredSource := []kapi.EnvVar{}
+	for _, env := range source {
+		for _, acceptable := range buildapi.WhitelistEnvVarNames {
```

Comment:
Would there ever be a case where we want to apply a whitelist or filter on the initial version of the output list provided?